pkg/apis/serving/v1alpha1/kfservice_types.go

```diff
@@ -42,6 +41,31 @@ type ModelSpec struct {
 	XGBoost    *XGBoostSpec    `json:"xgboost,omitempty"`
 	SKLearn    *SKLearnSpec    `json:"sklearn,omitempty"`
 	PyTorch    *PyTorchSpec    `json:"pytorch,omitempty"`
+	// Optional Explain specification to add a model explainer next to the chosen predictor.
+	// In future v1alpha2 the above model predictors would be moved down a level.
+	Explain *ExplainSpec `json:"explain,omitempty"`
+}
+
+// ExplainSpec defines the arguments for a model explanation server
+type ExplainSpec struct {
+	// The following fields follow a "1-of" semantic. Users must specify exactly one spec.
+	Alibi  *AlibiExplainSpec `json:"alibi,omitempty"`
+	Custom *CustomSpec       `json:"custom,omitempty"`
+}
+
+// AlibiExplainSpec defines the arguments for configuring an Alibi Explanation Server
+type AlibiExplainSpec struct {
+	// The following fields follow a "1-of" semantic. Users must specify exactly one of Type or ExplainerURI.
+	// The kind of Alibi explanation server to create, e.g. AnchorTabular
+	Type string `json:"type"`
```

Comment:
Can you make an enum for this? We can always add values in a backwards compat way.